src/Cake.Common/Xml/XmlPeekAliases.cs

```diff
@@ -0,0 +1,113 @@
+using Cake.Core;
+using Cake.Core.Annotations;
+using Cake.Core.IO;
+
+namespace Cake.Common.Xml
+{
+    /// <summary>
+    /// Contains functionality related to XML XPath queries.
+    /// </summary>
+    [CakeAliasCategory("XML")]
+    public static class XmlPeekAliases
+    {
+        /// <summary>
+        /// Gets the value of a target node.
+        /// </summary>
+        /// <returns>The value found at the given XPath query.</returns>
+        /// <param name="context">The context.</param>
+        /// <param name="filePath">The target file.</param>
+        /// <param name="xpath">The xpath of the node to get.</param>
+        [CakeMethodAlias]
+        public static string XmlPeek(this ICakeContext context, FilePath filePath, string xpath)
+        {
+            return context.XmlPeek(filePath, xpath, new XmlPokeSettings());
+        }
+        
+        /// <summary>
+        /// Get the value of a target node.
+        /// </summary>
+        /// <returns>The value found at the given XPath query.</returns>
+        /// <param name="context">The context.</param>
+        /// <param name="filePath">The target file.</param>
+        /// <param name="xpath">The xpath of the nodes to set.</param>
+        /// <param name="settings">Additional settings to tweak Xml Poke behavior.</param>
+        [CakeMethodAlias]
+        public static string XmlPeek(this ICakeContext context, FilePath filePath, string xpath, XmlPokeSettings settings)
+        {
+            if (context == null)
+            {
+                throw new ArgumentNullException("context");
+            }
+            
+            if (filePath == null)
+            {
+                throw new ArgumentNullException("filePath");
+            }
+            
+            var file = context.FileSystem.GetFile(filePath);
+            if (!file.Exists)
+            {
+                throw new FileNotFoundException("Source File not found.", file.Path.FullPath);
+            }
+            
+            using (var memoryStream = new MemoryStream())
+            {
+                using (var fileStream = file.Open(FileMode.Open, FileAccess.Read, FileShare.None))
+                using (var xmlReader = XmlReader.Create(fileStream))
+                using (var xmlWriter = XmlWriter.Create(memoryStream))
+                {
+                    return XmlPeek(xmlReader, xpath, settings);
+                }
+            }
+        }
+        
+        /// <summary>
+        /// Gets the value of a target node.
+        /// </summary>
+        /// <returns>The value found at the given XPath query (or the first, if multiple eligible nodes are found).</returns>
+        /// <param name="source">The source xml to transform.</param>
+        /// <param name="xpath">The xpath of the nodes to set.</param>
+        /// <param name="settings">Additional settings to tweak Xml Poke behavior.</param>
+        private static string XmlPeek(XmlReader source, string xpath, XmlPokeSettings settings)
```

Comment:
Addressed in a [new commit](https://github.com/cake-build/cake/commit/56a211eda0da0ffede7ba4edaaf4a309edefd287).